.github/workflows/test.yaml

```diff
@@ -24,12 +24,48 @@ jobs:
           mkdir -p $GITHUB_WORKSPACE/bin
           $GITHUB_WORKSPACE/scripts/download_geth_linux.py --dir $GITHUB_WORKSPACE/bin
           echo $GITHUB_WORKSPACE/bin >> $GITHUB_PATH
-      - name: Run Tox (PyPy)
+      - name: Run Tox Isort(PyPy)
         if: ${{ matrix.python == 'pypy-3.7' }}
-        run: tox -e pypy3
-      - name: Run Tox (CPython)
+        run: tox -e pypy3-isort
+      - name: Run Tox Flake8(PyPy)
+        if: ${{ matrix.python == 'pypy-3.7' }}
+        run: tox -e pypy3-flake8
+      - name: Run Tox Spec Lint(PyPy)
+        if: ${{ matrix.python == 'pypy-3.7' }}
+        run: tox -e pypy3-spec-lint
+      - name: Run Tox Frontier Tests(PyPy)
+        if: ${{ matrix.python == 'pypy-3.7' }}
+        run: tox -e pypy3-core-tests-frontier
+      - name: Run Tox Frontier Tests(PyPy)
+        if: ${{ matrix.python == 'pypy-3.7' }}
+        run: tox -e pypy3-core-tests-homestead
+      - name: Run Tox Frontier Tests(PyPy)
+        if: ${{ matrix.python == 'pypy-3.7' }}
+        run: tox -e pypy3-core-tests-tangerine-whistle
+      - name: Run Tox Isort(CPython)
+        if: ${{ matrix.python != 'pypy-3.7' }}
+        run: tox -e py3-isort
+      - name: Run Tox Black(CPython)
+        if: ${{ matrix.python != 'pypy-3.7' }}
+        run: tox -e py3-black
+      - name: Run Tox Flake8(CPython)
+        if: ${{ matrix.python != 'pypy-3.7' }}
+        run: tox -e py3-flake8
+      - name: Run Tox Mypy(CPython)
+        if: ${{ matrix.python != 'pypy-3.7' }}
+        run: tox -e py3-mypy
+      - name: Run Tox Spec Lint(CPython)
+        if: ${{ matrix.python != 'pypy-3.7' }}
+        run: tox -e py3-spec-lint
+      - name: Run Tox Frontier Tests(CPython)
+        if: ${{ matrix.python != 'pypy-3.7' }}
+        run: tox -e py3-core-tests-frontier
+      - name: Run Tox Frontier Tests(CPython)
+        if: ${{ matrix.python != 'pypy-3.7' }}
+        run: tox -e py3-core-tests-homestead
+      - name: Run Tox Frontier Tests(CPython)
         if: ${{ matrix.python != 'pypy-3.7' }}
-        run: tox -e py3
+        run: tox -e py3-core-tests-tangerine-whistle
```

Comment:
The `ethash` tests were failing. My conjecture is that since the tests are running in parallel, there's a high chance that the `ethhash` tests for different hardforks are competing for memory causing some tests to crash. So this change was more of an experiment to see if tests would pass if I would split tests for each hardfork as a separate tasks(and they did pass).

However, there's a problem with this approach. The coverage report is inaccurate since tests for each hard fork run independently. I need to get rid of this change and fix this issue in a separate PR. Sorry, for not informing earlier that this change was more of an experiment.